.github/workflows/build-cross-compile.yml

```diff
@@ -0,0 +1,142 @@
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+name: 'Build (cross-compile)'
+
+on:
+  workflow_call:
+    inputs:
+      apt-gcc-version:
+        required: false
+        type: string
+        default: '10=10.3.0-1ubuntu1~20.04'
+
+jobs:
+  build-cross-compile:
+    name: build
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        target-cpu:
+          - aarch64
+          - arm
+          - s390x
+          - ppc64le
+        include:
+          - target-cpu: aarch64
+            debian-arch: arm64
+            gnu-arch: aarch64
+          - target-cpu: arm
+            debian-arch: armhf
+            gnu-arch: arm
+            gnu-abi: eabihf
+          - target-cpu: s390x
+            debian-arch: s390x
+            gnu-arch: s390x
+          - target-cpu: ppc64le
+            debian-arch: ppc64el
+            gnu-arch: powerpc64le
+
+    steps:
+      - name: 'Checkout the JDK source'
+        uses: actions/checkout@v3
+
+      - name: 'Get the BootJDK'
+        id: bootjdk
+        uses: ./.github/actions/get-bootjdk
+        with:
+          platform: linux-x64
+
+        # Use linux-x64 JDK bundle as build JDK
+      - name: 'Get build JDK'
+        id: buildjdk
+        uses: ./.github/actions/get-bundles
+        with:
+          platform: linux-x64
+
+        # Upgrading apt to solve libc6 installation bugs, see JDK-8260460.
+      - name: 'Install toolchain and dependencies'
+        run: |
+          # Install dependencies using apt-get
+          sudo apt-get update
+          sudo apt-get install --only-upgrade apt
+          sudo apt-get install gcc-${{ inputs.apt-gcc-version }} g++-${{ inputs.apt-gcc-version }} libxrandr-dev${{ inputs.apt-architecture }} libxtst-dev${{ inputs.apt-architecture }} libcups2-dev${{ inputs.apt-architecture }} libasound2-dev${{ inputs.apt-architecture }}
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100 --slave /usr/bin/g++ g++ /usr/bin/g++-10
+          sudo apt-get install gcc-10-${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-abi}}=10.3.0-1ubuntu1~20.04cross1 g++-10-${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-abi}}=10.3.0-1ubuntu1~20.04cross1
```

Comment:
Should `10.3.0-1ubuntu1~20.04cross1` also go into common "var", like `apt-gcc-version` did?

Reply:
Also, I think we can speed up this part by merging two `apt-get install` invocation lines together. It was separate before, because it was two steps, unnecessary now.

Reply:
Ideally, all version information should be centralized somewhere. I plan to take a second round on this code when it's been pushed to try to figure out the best way to achieve this. (Possibly in github-versions.conf, possibly in main.yml, depending on what makes the most sense given the GHA limitations). 

But I can extract out the cross-compilation suffix already in this PR, yes.

Reply:
I can merge the two `apt-get install` lines, if you say that it is not necessary to call `update-alternatives` before the second install line. (But does it really speed things up?)

Reply:
> (But does it really speed things up?)

Yes, I think it does: `apt` would read the package database once, and do the post-install actions once. `update-alternatives` does not have to happen between those two lines.

Reply:
I hope I made the abstractions of the apt version strings correct. If not, we'll have to adjust it when we upgrade compilers.